From a982e3d5e3f529eb00d47a68119fc2b2e57fbae0 Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@ezurio.com>
Date: Mon, 28 Apr 2025 15:03:07 +0200
Subject: [PATCH 2/2] media: imx: isi-cap: fix subdev stream default value

Issue reported by Arducam where this parameter can be wrongly set to a
value other than 0 which makes the pipeline fail.

Signed-off-by: Gary Bisson <gary.bisson@ezurio.com>

Upstream-Status: Inappropriate NXP proprietary driver
---
 drivers/staging/media/imx/imx8-isi-cap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/staging/media/imx/imx8-isi-cap.c b/drivers/staging/media/imx/imx8-isi-cap.c
index 1c4c7c5a2fcf..053e1b4c06b1 100644
--- a/drivers/staging/media/imx/imx8-isi-cap.c
+++ b/drivers/staging/media/imx/imx8-isi-cap.c
@@ -972,6 +972,7 @@ static int mxc_isi_source_fmt_init(struct mxc_isi_cap_dev *isi_cap)
 	src_fmt.format.code = MEDIA_BUS_FMT_UYVY8_1X16;
 	src_fmt.format.width = dst_f->width;
 	src_fmt.format.height = dst_f->height;
+	src_fmt.stream = 0;
 	ret = v4l2_subdev_call(src_sd, pad, set_fmt, NULL, &src_fmt);
 	if (ret < 0 && ret != -ENOIOCTLCMD) {
 		v4l2_err(&isi_cap->sd, "set remote fmt fail!\n");
-- 
2.53.0
